testsuites/syncgateway/functional/tests/test_longpoll.py

```diff
@@ -131,3 +139,747 @@ def test_longpoll_changes_sanity(params_from_base_test_setup, sg_conf_name, num_
+        andy_changes_task = ex.submit(client.get_changes, url=sg_url, db=sg_db, since=andy_changes["last_seq"], timeout=10, auth=andy_auth)
+
+        # Wait for changes feed to notice there are no changes and enter wait. 2 seconds should be more than enough
+        time.sleep(2)
+
+        # Add add a doc for adam with "NBC" and "ABC" channels
+        # Add one doc, this should wake up the changes feed
+        adam_add_docs_task = ex.submit(client.add_docs, url=sg_url, db=sg_db,
+                                       number=1, id_prefix="adam_doc",
+                                       auth=adam_auth, channels=adam_user_info.channels)
+
+        # Wait for docs adds to complete
+        adam_docs = adam_add_docs_task.result()
+        assert len(adam_docs) == 1
+
+        # Assert that the changes feed woke up and that the doc change was propagated
+        adam_changes = adam_changes_task.result()
+        assert len(adam_changes["results"]) == 1
+        assert adam_changes["results"][0]["id"] == doc_id
+
+        # Verify that the changes feed is still listening for Traun and Andy
+        assert not traun_changes_task.done()
+        assert not andy_changes_task.done()
+
+        # Update the traun and andy to have one of adam's channels
+        update_traun_user_task = ex.submit(client.update_user, url=sg_admin_url, db=sg_db,
+                                           name=traun_user_info.name, password=traun_user_info.password, channels=["NBC"])
+        traun_auth = update_traun_user_task.result()
+
+        update_andy_user_task = ex.submit(client.update_user, url=sg_admin_url, db=sg_db,
+                                          name=andy_user_info.name, password=andy_user_info.password, channels=["ABC"])
+        andy_auth = update_andy_user_task.result()
+
+        # Make sure changes feed wakes up and contains at least one change
+        traun_changes = traun_changes_task.result()
+        assert 1 <= len(traun_changes["results"]) <= 2
+        changed_doc_ids = [change["id"] for change in traun_changes["results"]]
+        log_info("Changed doc_ids: {}".format(changed_doc_ids))
+        assert "adam_doc_0" in changed_doc_ids or "_user/traun" in changed_doc_ids
+
+        andy_changes = andy_changes_task.result()
+        assert 1 <= len(andy_changes["results"]) <= 2
+        changed_doc_ids = [change["id"] for change in andy_changes["results"]]
+        log_info("Changed doc_ids: {}".format(changed_doc_ids))
+        assert "adam_doc_0" in changed_doc_ids or "_user/andy" in changed_doc_ids
+
+    # Block until user docs are seen
+    client.verify_doc_id_in_changes(url=sg_url, db=sg_db, expected_doc_id="_user/adam", auth=adam_auth)
+    client.verify_doc_id_in_changes(url=sg_url, db=sg_db, expected_doc_id="_user/traun", auth=traun_auth)
+    client.verify_doc_id_in_changes(url=sg_url, db=sg_db, expected_doc_id="_user/andy", auth=andy_auth)
```

Comment:
The _user doc showing up on the changes feed isn't really a feature - it's more of a side-effect.  In particular, I can imagine us trying to find a better way to accomplish the same result in future (avoid duplicate entries after doing a channel backfill) without sending the _user doc.

Based on that, I'm on the fence a bit about whether we should code the tests this way - seems like they have the potential to break in future while the SG functionality is still correct.  

As discussed in the context of the recent bug #2186, a better validation might be to end the test by running a one-shot changes request with since=last_seq, and verify that it's not returning any rows.